aten/src/ATen/core/type.cpp

```diff
@@ -199,13 +199,20 @@ bool isSubvalueOf(const IValue& ivalue, TypePtr type) {
   return incompleteInferTypeFrom(ivalue)->isSubtypeOf(type);
 }
 
-
-c10::optional<TypePtr> unifyTypes(const TypePtr& t1, const TypePtr& t2) {
-  //cases that t1 == t2, or t1 is a type refinement of t2 and vice versa
+c10::optional<TypePtr> tryFindSuperType(const TypePtr& t1, const TypePtr& t2) {
```

Comment:
This name is somewhat misleading, because if we ever (or maybe we already) have a tree-like type hierarchy those checks would get insufficient (you'd need to find the least common ancestor).

Reply:
Ya i wasn't really sure what to name it...

Reply:
It's used twice so keeping it in a function makes sense, `whichOneIsTheSuperType` might be clearer name.